### PR TITLE
Story/epgr 21

### PR DESCRIPTION
**Pull request by Mihika0409**

### CODE REVIEW

File: grazer/src/main/java/edu/asu/diging/grazer/config/SecurityContext.java [hunk 1]
```diff
@@ -38,10 +43,10 @@ protected void configure(HttpSecurity http) throws Exception {
                 .and()
                 .authorizeRequests()
                 // Anyone can access the urls
-                .antMatchers("/", "/persons/**", "/person/**", "/concept/**", "/query**", "/info", "/resources/**", "/login/authenticate", "/rest/**",
+                .antMatchers("/", "/persons/**", "/person/**", "/concept/**", "/query**", "/info", "/resources/**", "/login/", "/loginfailed", "/rest/**",
                         "/logout").permitAll()
                 // The rest of the our application is protected.
-                .antMatchers("/users/**", "/admin/**", "/api/**").hasRole("ADMIN")
+                .antMatchers("/users/**", "/admin/**").hasRole("ADMIN")
```
Comment (jdamerow) [inline grazer/src/main/java/edu/asu/diging/grazer/config/SecurityContext.java]: we need to keep the api

File: grazer/src/main/java/edu/asu/diging/grazer/web/api/StartTransformationApiController.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
     @Autowired
     private TransformationCronJob cronJob;
 
-    @RequestMapping("/api/transformations/start")
+    @RequestMapping("/admin/api/transformations/start")
```
Comment (jdamerow) [inline grazer/src/main/java/edu/asu/diging/grazer/web/api/StartTransformationApiController.java]: let's keep the url as they are

File: grazer/src/main/webapp/WEB-INF/tiles/skeleton.jsp [hunk 3]
```diff
@@ -57,58 +58,97 @@
...
+            <div class="container">
+              <div class="row" style="padding-top: 15px;">
+                Welcome <i class="fa fa-user"></i><span style="margin-left: 5px;"><sec:authentication property="principal.username" /></span>!
+                <form action="<c:url value='/logout' />" method='POST' class="pull-right">
+                  <input type="hidden" name="${_csrf.parameterName}" value="${_csrf.token}"/>
+                  <div>
+                    <button type="submit" class="btn btn-link" style="color:#800000"><i class="fa fa-sign-out" aria-hidden="true"></i> Logout</button>
+                  </div>
+                </form>
+                <button type="button" class="btn btn-link pull-right"><a href="${pageContext.servletContext.contextPath}/admin/api/transformations/start">Start cron job</a></button>
```
Comment (jdamerow) [inline grazer/src/main/webapp/WEB-INF/tiles/skeleton.jsp]: Let's call this "Start Data Import"

Comment (jdamerow) [reply grazer/src/main/webapp/WEB-INF/tiles/skeleton.jsp]: what happens after this is called? what page is shown?

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: grazer/src/main/webapp/WEB-INF/tiles/skeleton.jsp (hunk 3 above)
Comment (Mihika0409) [reply grazer/src/main/webapp/WEB-INF/tiles/skeleton.jsp]: Once this is called, a page displays "Transformations started".

Comment (jdamerow) [reply grazer/src/main/webapp/WEB-INF/tiles/skeleton.jsp]: just as a text? this should show a full page like any other page of the application.

File: grazer/src/main/webapp/WEB-INF/tiles/skeleton.jsp [hunk 4]
```diff
@@ -57,58 +58,97 @@
...
+            <div class="container">
+              <div class="row" style="padding-top: 15px;">
+                Welcome <i class="fa fa-user"></i><span style="margin-left: 5px;"><sec:authentication property="principal.username" /></span>!
+                <form action="<c:url value='/logout' />" method='POST' class="pull-right">
+                  <input type="hidden" name="${_csrf.parameterName}" value="${_csrf.token}"/>
+                  <div>
+                    <button type="submit" class="btn btn-link" style="color:#800000"><i class="fa fa-sign-out" aria-hidden="true"></i> Logout</button>
+                  </div>
+                </form>
+                <button type="button" class="btn btn-link pull-right"><a href="${pageContext.servletContext.contextPath}/api/transformations/start">Start Data Import</a></button>
```
Comment (jdamerow) [inline grazer/src/main/webapp/WEB-INF/tiles/skeleton.jsp]: let's also show a warning dialog after the user clicks that button that warns them that when they proceed previously imported data will be deleted and asks if they want to proceed.

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: grazer/src/main/java/edu/asu/diging/grazer/web/api/StartTransformationApiController.java [hunk 5]
```diff
@@ -19,11 +19,19 @@
...
     private TransformationCronJob cronJob;
 
-    @RequestMapping("/api/transformations/start")
+    /*@RequestMapping("/api/transformations/start")
     public ResponseEntity<String> kickoffTransformations(@RequestParam(value="excludes", defaultValue="") String excludes) {
         List<String> excludeConcepts = Arrays.asList(excludes.split(","));
         excludeConcepts = excludeConcepts.stream().map(c -> c.trim()).collect(Collectors.toList());
         cronJob.retrieveTransformations(excludeConcepts);
         return new ResponseEntity<String>("Transformations started.", HttpStatus.OK);
+    }*/
```
Comment (jdamerow) [inline grazer/src/main/java/edu/asu/diging/grazer/web/api/StartTransformationApiController.java:28]: we need to keep the api

File: grazer/src/main/webapp/WEB-INF/views/dataImport.jsp [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<%@ page pageEncoding="UTF-8" %>
+<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c"%>
+<%@ taglib prefix="sec" uri="http://www.springframework.org/security/tags" %>
+<%@ taglib prefix="fn" uri="http://java.sun.com/jsp/jstl/functions" %>
+
+<h3>The import for the transformations has started...</h3>
```
Comment (jdamerow) [inline grazer/src/main/webapp/WEB-INF/views/dataImport.jsp:6]: Let's make that "Data import has started..."

File: grazer/src/main/webapp/WEB-INF/tiles/skeleton.jsp [hunk 7]
```diff
@@ -57,58 +58,115 @@
...
+                  </div>
+                </form>
+                <button type="button" class="btn btn-link pull-right" data-toggle="modal" data-target="#dataImportModal">Start Data Import</a></button>
+                <div class="modal fade" id="dataImportModal" role="dialog">
+                    <div class="modal-dialog">
+                    <!-- Modal content-->
+                        <div class="modal-content">
+                            <div class="modal-header">
+                                <button type="button" class="close" data-dismiss="modal">&times;</button>
+                                <h4 class="modal-title">Data Import</h4>
```
Comment (jdamerow) [inline grazer/src/main/webapp/WEB-INF/tiles/skeleton.jsp:81]: Let's make this "Import data"

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Pull request closed without merging**